flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/table/catalog/hive/util/HiveStatsUtil.java

```diff
@@ -90,6 +102,257 @@ public static Map<String, CatalogColumnStatisticsDataBase> createCatalogColumnSt
+                    long falseCount = 0L;
+                    Long nullCount = 0L;
+                    if (partitionValue == null) {
+                        nullCount =
+                                getNullCount(
+                                        client, hiveTable, partitionColIndex, defaultPartitionName);
+                    } else {
+                        Boolean boolVal = (Boolean) partitionValue;
+                        if (boolVal) {
+                            trueCount = 1L;
+                        } else {
+                            falseCount = 1L;
+                        }
+                    }
+                    return new CatalogColumnStatisticsDataBoolean(trueCount, falseCount, nullCount);
+                }
+            case TINYINT:
+            case SMALLINT:
+            case INTEGER:
+            case BIGINT:
+                {
+                    Long min = null;
+                    Long max = null;
+                    Long nullCount = 0L;
+                    if (partitionValue == null) {
+                        nullCount =
+                                getNullCount(
+                                        client, hiveTable, partitionColIndex, defaultPartitionName);
+                    } else {
+                        min = ((Number) partitionValue).longValue();
+                        max = min;
+                    }
+                    return new CatalogColumnStatisticsDataLong(min, max, 1L, nullCount);
+                }
+            case FLOAT:
+            case DOUBLE:
+            case DECIMAL:
+                {
+                    Double min = null;
+                    Double max = null;
+                    Long nullCount = 0L;
+                    if (partitionValue == null) {
+                        nullCount =
+                                getNullCount(
+                                        client, hiveTable, partitionColIndex, defaultPartitionName);
+                    } else {
+                        min = ((Number) partitionValue).doubleValue();
+                        max = min;
+                    }
+                    return new CatalogColumnStatisticsDataDouble(min, max, 1L, nullCount);
+                }
+            case DATE:
+                {
+                    Date min = null;
+                    Date max = null;
+                    Long nullCount = 0L;
+                    if (partitionValue == null) {
+                        nullCount =
+                                getNullCount(
+                                        client, hiveTable, partitionColIndex, defaultPartitionName);
+                    } else {
+                        min = new Date(((LocalDate) partitionValue).toEpochDay());
+                        max = min;
+                    }
+                    return new CatalogColumnStatisticsDataDate(min, max, 1L, nullCount);
+                }
+            default:
+                return null;
```

Comment:
TIMESTAMP is unsupported ?

Reply:
Miss it. `TIMESTAMP` should be supported.